node.go

```diff
@@ -59,8 +67,12 @@ func SetNodeInterface(name string) bool {
 // NodeID returns a slice of a copy of the current Node ID, setting the Node ID
 // if not already set.
 func NodeID() []byte {
+	lock.RLock()
+	defer lock.RUnlock()
 	if nodeID == nil {
+		lock.RUnlock()
 		SetNodeInterface("")
```

Comment:
Sorry for the delay in review.  For some reason I am not seeing the mail from github most of the time.

In any event, this change looks good except here.  There is a race between releasing the read lock and calling SetNodeInterface.  Have you seen performance issues with just using a regular lock here?  In general this is a pretty quick operation (except the one time that SetNodeInterface is called) so I think a regular mutex would be fine.  SetNodeInterface should become:

func SetNodeInterface(name string) bool {
    lock.Lock()
   defer logk.Unlock()
   return setNodeInterface(name)
}

func SetNodeInterface(name string) bool {
....

And then call setNodeInterface here with the lock held.